plugins/search-indexer-backend/src/index.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Registry } from './registry';
+import {
+  RegisterCollatorParameters,
+  RegisterDecoratorParameters,
+} from './types';
+
+const registry = Registry.getInstance();
+export const registerCollator = (params: RegisterCollatorParameters) => {
+  registry.addCollator(params);
+};
+export const registerDecorator = (params: RegisterDecoratorParameters) => {
+  registry.addDecorator(params);
+};
+
+export type {
+  IndexableDocument,
+  IndexableDocumentCollator,
+  IndexableDocumentDecorator,
+} from './types';
```

Comment:
Finding the existence of this as a separate backend plugin a bit weird tbh, since it expose any routes. Perhaps we can use https://backstage.io/docs/architecture-decisions/adrs-adr011 and put this in a `search-backend-node` package instead? Or just straight in the `search-backend`

Reply:
I'm open to just merging it into `search-backend`.  This came out of [a note in the architecture feedback](https://github.com/backstage/backstage/issues/4078#issuecomment-767382529) about other backend plugins being able to just depend on the indexing bits without having to depend on the querying bits.  I could kinda go either way.  Whaddya think?

I do think it's important to firmly expand the scope of common responsibilities in backend plugins beyond exposing routes though.  Declaring and running async/background tasks seems like it should be a first-class concern for a backend plugin.